Transformers/Java/src/main/java/com/github/ciselab/lampion/transformations/transformers/RenameVariableTransformer.java

```diff
@@ -0,0 +1,242 @@
+
+
+    public RenameVariableTransformer() {
+        super();
+        setConstraints();
+    }
+
+    public RenameVariableTransformer(long seed) {
+        super(seed);
+        setConstraints();
+    }
+
+    /**
+     * This method applied the class-specific Transformation to a random, valid element of the given AST.
+     * It should check itself for constraints given.
+     * <p>
+     * The Transformation returns a TransformationResult-Element, that holds all relevant information.
+     * In case of a failing transformation or unmatched constraints, return an EmptyTransformationResult.
+     * <p>
+     * The AST is altered in the process.
+     *
+     * @param ast The toplevel AST from which to pick a qualified children to transform.
+     * @return The TransformationResult, containing all relevant information of the transformation
+     */
+    @Override
+    public TransformationResult applyAtRandom(CtElement ast) {
+        // Sanity check, if there are blockers in the constraints return empty TransformationResult
+        if (!getRequirements().stream().allMatch(r -> r.test(ast))) {
+            return new EmptyTransformationResult();
+        }
+        Optional<CtMethod> oToAlter = pickRandomMethod(ast);
+        // Check for existance of methods is done beforehand per constraints, so I just get the result right away
+        CtMethod toAlter = oToAlter.get();
+
+        Optional<CtLocalVariable> oVarToAlter = pickRandomVariable(toAlter);
+        // As the altered method is altered forever and in all instances, safe a clone for the transformation result.
+        CtMethod savedElement = toAlter.clone();
+        savedElement.setParent(toAlter.getParent());
+        savedElement.getParent().updateAllParentsBelow();
+
+        applyRenameVariableTransformer(toAlter, oVarToAlter.get());
+
+        // If debug information is wished for, create a bigger Transformationresult
+        // Else, just return a minimal Transformationresult
+        if (debug) {
+            return new SimpleTransformationResult(name,savedElement,this.getCategories(),beforeAfterOverview(savedElement,toAlter),ast.clone());
+        } else {
+            return new SimpleTransformationResult(name,savedElement,this.getCategories());
+        }
+    }
+
+    private void applyRenameVariableTransformer(CtMethod toAlter, CtLocalVariable varToAlter) {
+        CtRenameLocalVariableRefactoring refac = new CtRenameLocalVariableRefactoring();
+        //CtRenameGenericVariableRefactoring refac = new CtRenameGenericVariableRefactoring();
+        refac.setTarget(varToAlter);
+        String name = fullRandomStrings ? RandomNameFactory.getRandomString(random) : RandomNameFactory.getCamelcasedAnimalString(false,random);
+        refac.setNewName(name);
+        refac.refactor();
+
+        // Add the altered variable to the toplevel map to keep track that it was altered in constraints
+        if(alreadyAlteredVariableNames.containsKey(toAlter)){
```

Comment:
I think alreadyAlteredVariableNames is a bit unlucky, 
I do not like the "already". Maybe "known" or "seen"